base/cvd/cuttlefish/host/libs/image_aggregator/detect_image.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "cuttlefish/common/libs/utils/result.h"
+#include "cuttlefish/host/libs/image_aggregator/disk_image.h"
+
+namespace cuttlefish {
+
+Result<std::unique_ptr<DiskImage>> DetectImage(const std::string& path);
```

Comment:
nit: I'd expect a function named `DetectImage` to return the image type or something like that, but this return type looks like a full image instead of just a type. Consider `ImageFromFile(...)` or `DiskImage::FromFile(...)`.

Reply:
Renamed to `ImageFromFile`.

Unfortunately this can't be in the `DiskImage` class as it would create a circular dependency between the `:disk_image` bazel target and the subclass bazel targets.